docs/modules/pkl-cli/pages/index.adoc

```diff
@@ -827,3 +827,18 @@ These certificates can be overridden via either of the two options:
 Both these options will *replace* the default CA certificates bundled with Pkl. +
 The CLI option takes precedence over the certificates in `~/.pkl/cacerts/`. +
 Certificates need to be X.509 certificates in PEM format.
+
+[[http-proxy]]
+== HTTP Proxy
+
+When making HTTP(S) requests, Pkl can use a proxy.
+By default, no proxy is configured.
+A proxy can be configured as follows:
+
+- Using `--proxy <uri>`.
+  The URI must (currently) have scheme `http`, and may not contain anything other than a host and port.
+- Using `--no-proxy <hosts>`.
+  The given (comma separated list of) hosts bypass the proxy.
+  Hosts can be specified by domain name (in which case all subdomains also bypass the proxy), IP addresses, or IP ranges (using link:https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing#CIDR_notation[CIDR notation]).
+  For individual hosts (not CIDRs), ports can be specified.
+  When no port is specified for a given host, connections to all ports bypass the proxy.
```

Comment:
By default, Pkl actually uses the system proxy settings.

I think something that's missing here is some documentation about how Pkl resolves proxy settings. E.g. the stuff about looking in the CLI flags, then PklProject, then the settings file.

BTW: the language here describes the settings as CLI flags, which is a little misleading (no proxy within `settings.pkl` isn't a comma-separated list).